src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+                return Task.CompletedTask;
+            }
+
+            bool ShouldSendCloseResponseOnClose(out RequestContext pendingCloseRequestContext, out Message pendingCloseResponse)
+            {
+                bool sendCloseResponse = false;
+                lock (ThisLock)
+                {
+                    this.canSendCloseResponse = true;
+                    if (!this.sentCloseResponse && this.receivedClose && this.closeResponse != null)
+                    {
+                        this.sentCloseResponse = true;
+                        sendCloseResponse = true;
+                        pendingCloseRequestContext = this.closeRequestContext;
+                        pendingCloseResponse = this.closeResponse;
+                        this.closeResponse = null;
+                        this.closeRequestContext = null;
+                    }
+                    else
+                    {
+                        canSendCloseResponse = false;
+                        pendingCloseRequestContext = null;
+                        pendingCloseResponse = null;
+                    }
+                }
+                return sendCloseResponse;
+            }
+
+            bool SendCloseResponseOnCloseIfRequired(CancellationToken token)
+            {
+                bool aborted = false;
+                RequestContext pendingCloseRequestContext;
+                Message pendingCloseResponse;
+                bool sendCloseResponse = ShouldSendCloseResponseOnClose(out pendingCloseRequestContext, out pendingCloseResponse);
+                bool cleanupCloseState = true;
+                if (sendCloseResponse)
+                {
+                    try
+                    {
+                        this.SendCloseResponse(pendingCloseRequestContext, pendingCloseResponse, token);
+                        // this.inputSessionClosedHandle.Set();
+                        cleanupCloseState = false;
+                    }
+                    catch (CommunicationObjectAbortedException)
+                    {
+                        if (this.State != CommunicationState.Closed)
+                        {
+                            throw;
+                        }
+                        aborted = true;
+                    }
+                    finally
+                    {
+                        if (cleanupCloseState)
+                        {
+                            if (pendingCloseResponse != null)
+                            {
+                                pendingCloseResponse.Close();
+                            }
+                            if (pendingCloseRequestContext != null)
+                            {
+                                pendingCloseRequestContext.Abort();
+                            }
+                        }
+                    }
+                }
+
+                return aborted;
+            }
+
+            //bool WaitForInputSessionClose(TimeSpan timeout, out bool wasAborted)
```

Comment:
Remove as even if mechanism needs reimplementing, it will look a lot different than this.